packages/core/test/template-engine/built-in-extensions/masking/masking.spec.ts

```diff
@@ -0,0 +1,120 @@
+
+  const { compiledData } = await compiler.compile(
+    `{% masking id partial(2, 'xxxxxxx', 0) %}`
+  );
+
+  // Action
+  loader.setSource('test', compiledData);
+  await executeTemplate('test');
+  const queries = await getCreatedQueries();
+
+  // Assert
+  expect(queries[0]).toBe(`concat(substr(id, 1, 2), 'xxxxxxx')`);
+});
+
+it('If an argument is not a symbol, extension should throw', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(compiler.compile(`{% masking id %}`)).rejects.toThrow(
+    `Expected a symbol, but got a block end`
+  );
+});
+
+it('If not the partial masking function, extension should throw', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`{% masking id fakeFunc(2, 'xxxxxxx', 2) %}`)
+  ).rejects.toThrow(`Expected the "partial" function name`);
+});
+
+it('If not correct use the masking function, the extension should throw', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(compiler.compile(`{% masking id partial %}`)).rejects.toThrow(
+    `Expected a function start`
+  );
+});
+
+it('If the first argument type of the masking function is incorrect, the extension should throw', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`{% masking id partial('2', 'xxxxxxx', 2) %}`)
+  ).rejects.toThrow(`Expected a int, but got string`);
+});
+
+it('If the second argument type of the masking function is incorrect, the extension should throw', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`{% masking id partial(2, xxxxxxx, '2') %}`)
+  ).rejects.toThrow(`Expected a string, but got symbol`);
+});
+
+it('If the third argument type of the masking function is incorrect, the extension should throw', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`{% masking id partial(2, 'xxxxxxx', '2') %}`)
+  ).rejects.toThrow(`Expected a int, but got string`);
+});
+
+it('If arguments of masking function are incorrect, extension should throw', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`{% masking id partial(2, 'xxxxxxx') %}`)
+  ).rejects.toThrow(`Expected a function has 3 arguments`);
+});
+
+it('If not correct use the masking function whole syntax, the extension should throw', async () => {
+  // Arrange
+  const { compiler } = await createTestCompiler();
+
+  // Action, Assert
+  await expect(
+    compiler.compile(`{% masking id partial(2, 'xxxxxxx', 3 %}`)
+  ).rejects.toThrow(`Expected a function end`);
+});
```

Comment:
NIT: We can add some test cases for parser position checking:
```sql
AA{% masking id partial(2, 'xxxxxxx', 0) %}AA
```

NIT: We can add some test cases for the toleration of spaces
```sql
{% masking id partial(2,                'xxxxxxx'                   , 3) %}
```